File: FFRv2/app/src/main/java/com/devingotaswitch/rankings/sources/ParseDraftWizard.java
package com.devingotaswitch.rankings.sources;

import android.util.Log;

import com.devingotaswitch.rankings.domain.Player;
import com.devingotaswitch.rankings.domain.Rankings;
import com.devingotaswitch.utils.JsoupUtils;
import com.devingotaswitch.utils.ParsingUtils;

import java.io.IOException;
import java.util.List;

public class ParseDraftWizard {

    public static void parseRanksWrapper(Rankings rankings)
            throws IOException {
        String type = "STD";
        if (rankings.getLeagueSettings().getScoringSettings().getReceptions() > 0) {
            type = "PPR";
        }
        String url = "http://draftwizard.fantasypros.com/editor/createFromProjections.jsp?sport=nfl&scoringSystem="
                + type + "&showAuction=Y";
        url += "&teams=" + rankings.getLeagueSettings().getTeamCount();
        url += "&QB=" + rankings.getLeagueSettings().getRosterSettings().getQbCount();
        url += "&WR=" + rankings.getLeagueSettings().getRosterSettings().getWrCount();
        url += "&RB=" + rankings.getLeagueSettings().getRosterSettings().getRbCount();
        url += "&TE=" + rankings.getLeagueSettings().getRosterSettings().getTeCount();
        url += "&DST=" + rankings.getLeagueSettings().getRosterSettings().getDstCount();
        url += "&K=" + rankings.getLeagueSettings().getRosterSettings().getkCount();
        if (rankings.getLeagueSettings().getRosterSettings().getFlex() != null) {
            url += "&WR/RB=" + rankings.getLeagueSettings().getRosterSettings().getFlex().getRbwrCount();
            url += "&WR/RB/TE=" + rankings.getLeagueSettings().getRosterSettings().getFlex().getRbwrteCount();
            url += "&RB/TE=" + rankings.getLeagueSettings().getRosterSettings().getFlex().getRbteCount();
            url += "&WR/TE=" + rankings.getLeagueSettings().getRosterSettings().getFlex().getWrteCount();
            url += "&QB/WR/RB/TE=" + rankings.getLeagueSettings().getRosterSettings().getFlex().getQbrbwrteCount();
        }
        parseRanksWorker(rankings, url);
    }

    public static void parseRanksWorker(Rankings rankings, String url)
            throws IOException {
        List<String> td = JsoupUtils.handleLists(url,
                "table#OverallTable td");
        Log.d("FP", url);
        int startingIndex = 0;
        for (int i = 0; i < td.size(); i++) {
            if (td.get(i).contains(" - ") && td.get(i).split(" ").length > 3) {
                startingIndex = i;
                break;
            }
        }
        for (int i = startingIndex; i < td.size(); i += 5) {
            double aucVal = Double.parseDouble(td.get(i + 2).substring(1,
                    td.get(i + 2).length()));
            String playerName = td.get(i).split(" \\(")[0];
            String teamPos = td.get(i).split(" \\(")[1];
            String team = teamPos.split(" - ")[0];
            String pos = teamPos.split(" - ")[1].split("\\)")[0];
            Player player = ParsingUtils.getPlayerFromRankings(playerName, team, pos, aucVal);

            // Double count it because math
            rankings.processNewPlayer(player);
            rankings.processNewPlayer(player);
        }
    }
}

File: FFRv2/app/src/main/java/com/devingotaswitch/rankings/sources/ParseDraftWizard.kt
package com.devingotaswitch.rankings.sources

import com.devingotaswitch.rankings.domain.Rankings
import com.devingotaswitch.utils.JsoupUtils.parseURLWithUA
import com.devingotaswitch.utils.ParsingUtils.getPlayerFromRankings
import java.io.IOException

object ParseDraftWizard {
    @Throws(IOException::class)
    fun parseRanksWrapper(rankings: Rankings) {
        var type = "STD"
        if (rankings.getLeagueSettings().scoringSettings.receptions >= 1.0) {
            type = "PPR"
        }
        if (rankings.getLeagueSettings().scoringSettings.receptions > 0) {
            type = "HALF"
        }
        var url = ("http://draftwizard.fantasypros.com/editor/createFromProjections.jsp?sport=nfl&scoringSystem="
                + type + "&showAuction=Y")
        url += "&teams=" + rankings.getLeagueSettings().teamCount
        url += "&tb=200"
        var qbTotal = rankings.getLeagueSettings().rosterSettings.qbCount
        if (rankings.getLeagueSettings().rosterSettings.flex != null) {
            qbTotal += rankings.getLeagueSettings().rosterSettings.flex!!.qbrbwrteCount
        }
        url += "&QB=$qbTotal"
        url += "&RB=" + rankings.getLeagueSettings().rosterSettings.rbCount
        url += "&WR=" + rankings.getLeagueSettings().rosterSettings.wrCount
        url += "&TE=" + rankings.getLeagueSettings().rosterSettings.teCount
        url += "&DST=" + rankings.getLeagueSettings().rosterSettings.dstCount
        url += "&K=" + rankings.getLeagueSettings().rosterSettings.kCount
        url += "&BN=" + rankings.getLeagueSettings().rosterSettings.benchCount
        // Removed the flex customizations, they were being dropped and a zero value provided
        parseRanksWorker(rankings, url)
    }

    @Throws(IOException::class)
    private fun parseRanksWorker(rankings: Rankings, url: String) {
        val td = parseURLWithUA(url,
                "table#OverallTable td")
        var startingIndex = 0
        for (i in td.indices) {
            if (td[i].contains(" - ") && td[i].split(" ").size > 3) {
                startingIndex = i
                break
            }
        }
        var i = startingIndex
        while (i < td.size) {
            val aucVal = td[i + 2].substring(1
            ).toDouble()
            val playerName = td[i].split(" \\(".toRegex())[0]
            val teamPos = td[i].split(" \\(".toRegex())[1]
            val team = teamPos.split(" - ")[0]
            val pos = teamPos.split(" - ")[1].split("\\)".toRegex())[0]
            val player = getPlayerFromRankings(playerName, team, pos, aucVal)

            rankings.processNewPlayer(player)
            i += 5
        }
    }
}